core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -547,7 +548,8 @@ public void retry(String workflowId) {
         if (workflow.getTasks().isEmpty()) {
             throw new ApplicationException(CONFLICT, "Workflow has not started yet");
         }
-
+        Optional<Task> lTask = workflow.getTasks().stream().filter(this::findLastFailedOrTimeOutTask).findFirst();
+        workflow = findLastFailedSubWorkflow(lTask.get(),workflow);
```

Comment:
@mehmet-islamoglu Thanks for the contribution. As mentioned in the previous comment, this change should be backwards compatible, ```I.e when retried from 
parent, with a flag to keep the existing behavior of restarting the subworkflow vs retrying it, which would recursively evaluate the subworkflow(s) until finding the last discrete task and retrying it (For eg., in a multi-level sub workflow).```.

Subworkflows will no longer be restarted with this change, which might break few users. Please update the PR to accommodate this.

Reply:
Hello @kishorebanala what is the reason behind keeping existing behavour if it is causing bugs?

Reply:
@azb96 Can you elaborate on the existing behavior causing bugs please? 

While it naturally makes sense to retry Subworkflow task on retrying the parent, it is possible that folks might've built their existing pipelines in such a way that the expectation is to recreate a new Subworkflow on retrying the parent. Hence the reason to keep this backwards compatible. We generally avoid removing the existing features in minor releases, and Conductor 3.0 effort might be a good place to re-consider this.